ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -96,20 +97,15 @@ protected void handleError(final Throwable error) {
   }
 
   protected void handleError(final Throwable error, final boolean couldBeAuthError) {
-    final long timeNow = timeProvider.getTimeInMillis().longValue();
-    final long maybeUpdatedTime =
-        lastError.accumulateAndGet(
-            timeNow,
-            (lastErrorTime, givenErrorTimeUpdate) -> {
-              if (lastErrorTime == NO_ERROR_TIME
-                  || givenErrorTimeUpdate - lastErrorTime > ERROR_REPEAT_DELAY_MILLIS) {
-                return givenErrorTimeUpdate;
-              }
-              return lastErrorTime;
-            });
-    if (maybeUpdatedTime == timeNow) {
-      logExecutionClientError(error, couldBeAuthError);
+    if (error instanceof SocketException) { // indicates that EL is offline
+      if (shouldReportError()) {
```

Comment:
I think it would be better to keep the `executionClientEventsPublisher.onAvailabilityUpdated(false)` in here to keep considering the `ERROR_REPEAT_DELAY_MILLIS`. In my understanding, we shouldn't consider the EL unavailable at the first error (There could some minor network issues or timeouts and that doesn't necessarily mean the EL isn't available)

Reply:
Agree, left only the logging changes, since I am not sure as well.